src/ui.data-table/UTableRow.vue

```diff
@@ -31,21 +31,27 @@
         :style="getNestedShift()"
         v-bind="bodyCellNestedAttrs"
       >
-        <UIcon
-          v-if="isShownToggleIcon"
-          size="xs"
-          internal
-          interactive
-          :name="getToggleIconName(row)"
-          color="brand"
-          v-bind="toggleIconConfig"
-          @click.stop="onClickToggleIcon"
-        />
-
+        <div
+          ref="toggle-wrapper"
+          v-bind="bodyCellNestedExpandIconWrapperAttrs"
+          :style="{ width: getIconWidth() }"
+        >
+          <UIcon
+            v-if="isShownToggleIcon"
+            size="xs"
+            internal
+            interactive
+            :data-row-toggle-icon="`row-toggle-icon-${row.id}`"
```

Comment:
maybe do it simpler:

`:data-toggle-icon-row="row.id"`

Reply:
It's still toggle button of the row, not the whole row, so I left the name